test/lib/custom-assertions.js

```diff
@@ -106,8 +106,11 @@ function compareSegments(parent, segments) {
  *                                  validation.  This is useful, for example, when a
  *                                  segment may or may not be created by code that is not
  *                                  directly under test.  Only used when `exact` is true.
+ * @param {object} [options] Set of optional parameters.
+ * @param {object} [options.assert] The assertion library to use. Defaults to
+ * `node:assert`.
  */
-function assertSegments(parent, expected, options) {
+function assertSegments(parent, expected, options, { assert = require('node:assert') } = {}) {
```

Comment:
I updated these function to accept an assertion library in order to work with `@matteo.collina/tspl`. In the Kafka suite, I found that using a plan was beneficial, but the tests utilized these helper functions. In order for the `plan` to work, we need to use the assertion methods exported by it. Thus, passing in the library to the functions.